base/iterators.jl

```diff
@@ -925,6 +931,9 @@ julia> collect(Iterators.product(1:2, 3:5))
 2×3 Matrix{Tuple{Int64, Int64}}:
  (1, 3)  (1, 4)  (1, 5)
  (2, 3)  (2, 4)  (2, 5)
+
+julia> ans == [(x,y) for x in 1:2, y in 3:5]  # lowers to the same code
```

Comment:
This is not true, I think.

While it lowers to very similar code (both use `Iterators.product`), it is not equivalent.  The comprehension version creates a `Generator` object, whereas collecting the iterator directly does not.

Reply:
Indeed that's not quite right, sorry, will fix. 

Literally showing what it is is probably too verbose, and not so easy to read... and will break `jldoctest` I presume. 
```
julia> ((x,y) for x in 1:2, y in 3:5)
Base.Generator{Base.Iterators.ProductIterator{Tuple{UnitRange{Int64}, UnitRange{Int64}}}, var"#65#66"}(var"#65#66"(), Base.Iterators.ProductIterator{Tuple{UnitRange{Int64}, UnitRange{Int64}}}((1:2, 3:5)))
```